rtt_rosparam/src/rtt_rosparam_service.cpp

```diff
@@ -239,7 +247,27 @@ class ROSParamService: public RTT::Service
   bool setParamComponentPrivate(const std::string &name) { return set(name, COMPONENT_PRIVATE); }
   bool setParamComponentRelative(const std::string &name) { return set(name, COMPONENT_RELATIVE); }
   bool setParamComponentAbsolute(const std::string &name) { return set(name, COMPONENT_ABSOLUTE); }
-};
+
+  /**
+   * Adds a ROS parameter of supported types by ROS param as a property to this
+   * bag.
+   * A Property is created which causes contents of the
+   * property always to be in sync
+   * with the contents of the ROS parameter.
+   * @param name The name of this property
+   * @return the Property object by reference, which you can further query or document.
+   */
+  template<typename T, ResolutionPolicy P>
+  Property<T>& addRosParamProperty( const std::string& name) {
+    const std::string resolved_name = resolvedName(name,P);
+    typename internal::AssignableDataSource<T>::shared_ptr datasource( new rtt_rosparam::RosParamDataSource<T>(resolved_name) );
+    Property<T>* p = new Property<T>(name,"", datasource);
+    p->doc("Property generated by rosparam attached to parameter: " + resolved_name);
```

Comment:
_Optional:_
```suggestion
    Property<T>* p = new Property<T>(name, "ROS parameter " + resolved_name, datasource);
```

Not sure why you first construct the `Property<T>` with an empty description and then set it afterwards with `p->doc()`?